src/App.tsx

```diff
@@ -1,26 +1,46 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
+const totalItems = items.length;
 
 export const App: React.FC = () => {
+  const [visiblePages, setVisiblePages] = useState(5);
+  const [selectedPage, setSelectedPage] = useState(1);
+
+  const startInd = selectedPage * visiblePages - visiblePages;
+  const endInd = Math.min(startInd + visiblePages, totalItems);
+
+  const handlerSelectedPage = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setVisiblePages(+event.target.value);
+    setSelectedPage(1);
+  };
+
+  const handlePageChange = (page: number) => {
+    if (page !== selectedPage) {
+      setSelectedPage(page);
+    }
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${selectedPage} (items ${startInd + 1} - ${endInd} of ${totalItems})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
+            value={visiblePages}
             data-cy="perPageSelector"
             id="perPageSelector"
             className="form-control"
+            onChange={handlerSelectedPage}
```

Comment:
The handler for `onChange` in the select element should be renamed to `handleVisiblePagesChange` for a clearer indication of its function.